From b9836c120a94c21f6d2eb8bc706408741b16b346 Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Fri, 10 Jan 2014 17:52:37 +0000
Subject: users: Improve logged-in check when deleting users

When deleting a user account, we try to check if the user is
still logged in, but we are using an accountsservice API that
ignores console logins. But deleting a user that is logged
in on the console is just as bad, so use
act_user_is_logged_in_anywhere instead, which includes console
logins.

https://bugzilla.gnome.org/show_bug.cgi?id=721951
---
diff --git a/panels/user-accounts/um-user-panel.c b/panels/user-accounts/um-user-panel.c
index 2c02520..058eaad 100644
--- panels/user-accounts/um-user-panel.c
+++ panels/user-accounts/um-user-panel.c
@@ -478,7 +478,7 @@ delete_user (GtkButton *button, CcUserPanelPrivate *d)
                 g_signal_connect (dialog, "response",
                                   G_CALLBACK (gtk_widget_destroy), NULL);
         }
-        else if (act_user_is_logged_in (user)) {
+        else if (act_user_is_logged_in_anywhere (user)) {
                 dialog = gtk_message_dialog_new (GTK_WINDOW (gtk_widget_get_toplevel (d->main_box)),
                                                  0,
                                                  GTK_MESSAGE_INFO,
--
cgit v0.9.2
